### PR TITLE
Move some docstrings from helpdb to function definitions

### PR DESCRIPTION
**Pull request by OliverEvans96**
Thanks for reviewing, I hope it's okay! :)

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -858,6 +858,16 @@ end
 # note: the following type definitions don't mean any AbstractArray is convertible to
 # a data Ref. they just map the array element type to the pointer type for
 # convenience in cases that work.
+
+"""
+    pointer(array [, index])
+
+Get the native address of an array or string element. Be careful to ensure that a Julia
+reference to `a` exists as long as this pointer will be used. This function is "unsafe" like
```
Comment (tkelman) [inline base/abstractarray.jl]: the variable name a isn't referenced in the signature, might make more sense if consistent

**Comment by KristofferC** [comment]
I wonder if some of these docstrings occur before enough of the documentation system itself has been loaded (which would explain the CI problems).

**Comment by fredrikekre** [comment]
https://github.com/JuliaLang/julia/pull/22521#issuecomment-310878205

**Comment by KristofferC** [comment]
So the easiest way forward would be to move some of the stuff (`boot.jl`) back into the helpdb file. You can check that everything works locally by rebuilding Julia.

File: base/io.jl [hunk 2]
```diff
@@ -419,6 +403,28 @@ function read(s::IO, ::Type{Char})
     return Char(c)
 end
 
+"""
+    read!(stream::IO, array::Union{Array, BitArray})
+    read!(filename::AbstractString, array::Union{Array, BitArray})
+
+Read binary data from an I/O stream or file, filling in `array`.
+"""
+@noinline function read!(s::IO, a::Array{T}) where T # mark noinline to ensure the array is gc-rooted somewhere (by the caller)
```
Comment (fredrikekre) [inline base/io.jl]: Just curious, any reason these functions where moved down? :)

File: base/error.jl [hunk 3]
```diff
@@ -65,6 +65,12 @@ systemerror(p, b::Bool; extrainfo=nothing) = b ? throw(Main.Base.SystemError(str
 
 ## assertion functions and macros ##
 
+"""
+    assert(cond)
+
+Throw an [`AssertionError`](@ref) if `cond` is `false`.
+Also available as the macro `@assert expr`.
```
Comment (fredrikekre) [inline base/error.jl]: perhaps only ``... as the macro [`@assert`](@ref).``? (given that `@assert` is documented)

File: base/array.jl [hunk 4]
```diff
@@ -212,6 +212,37 @@ function getindex(::Type{Any}, vals::ANY...)
...
 getindex(::Type{Any}) = Array{Any,1}(0)
 
+"""
+    fill!(A, x)
+
+Fill array `A` with the value `x`. If `x` is an object reference, all elements will refer to
+the same object. `fill!(A, Foo())` will return `A` filled with the result of evaluating
+`Foo()` once.
+
+```jldoctest
```
Comment (fredrikekre) [inline base/array.jl]: While you are at it, perhaps add an `# Examples` heading before the doctests (as in #22522)? :)

**Pull request closed without merging**